### ISSUE TITLE
TypeError: '<' not supported between instances of 'FzLocation' and 'int' in doc.delete_pages

### ISSUE THREAD

**Issue by Kaschi14** [pymupdf/PyMuPDF#3094]
### Description of the bug

When trying to doc.delete_pages(pages_to_rm) I get the error:

```
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "path/to/src/split_pdfs.py", line 172, in main
    pdf_splitter_manu.process_pdf()
  File "path/to/src/split_pdfs.py", line 158, in process_pdf
    self.remove_images()
  File "path/to/src/split_pdfs.py", line 49, in remove_images
    self.doc.delete_pages(pages_to_rm)
  File "path/to/.venv/lib/python3.9/site-packages/fitz/__init__.py", line 3715, in delete_pages
    self._remove_links_to(frozen_numbers)
  File "path/to/.venv/lib/python3.9/site-packages/fitz/__init__.py", line 3330, in _remove_links_to
    _remove_dest_range(pdf, numbers)
  File "path/to/.venv/lib/python3.9/site-packages/fitz/__init__.py", line 14095, in _remove_dest_range
    if pno < 0: # page number lookup did not work
TypeError: '<' not supported between instances of 'FzLocation' and 'int'
"""
```

### How to reproduce the bug

Unfortunately I cannot share the PDF as it is sensitive.

But this is the function I use:

```
def remove_images(self):
        pages_to_rm = []
        # self.doc.set_named_dests({})
        for page_num in range(self.doc.page_count):
            page = self.doc[page_num]

            # for link in page.get_links():
            #     page.delete_link(link)

            # Check if page has any text, if not delete page and continue
            if not page.get_text().strip():
                pages_to_rm.append(page_num)
                continue

            # Remove images
            images = page.get_images(full=True)

            for img_index, img_info in enumerate(images):
                try:
                    page.delete_image(img_info[0])
                except ValueError:
                    continue

        print(pages_to_rm)
        self.doc.delete_pages(pages_to_rm)
        self.doc.save(self.output_path)
        self.doc.close()
```

When tracing the error it looked like it might be a problem with links / bookmarks, thats why I tried to remove all links. But when doing this my script is stuck at some point without any error.

### PyMuPDF version

1.23.16

### Operating system

MacOS

### Python version

3.9

**Comment by julian-smith-artifex-com** [comment]
Actually we've managed to reproduce the problem, and i have a fix in my tree.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.19.
